### ISSUE TITLE
iOS: iPad 12" browser view size issue

### ISSUE THREAD

**Issue by samuelclay** [samuelclay/NewsBlur#1154]
Email from a user. I believe we may have fixed this but if not I figure it'll be quick.

--
Hi, I’ve noticed this intermittent issue when opening links in the app on the 2018 12.9” iPad Pro. The article will open with a black box on the right, and after closing the article, the display of the feed will look like the attached until the app is restarted. I haven’t been able to figure out what the articles that trigger this have in common, but I’d say it happens about 5-10% of the time I click on a link.

Sorry this is vague! Let me know how I can help troubleshoot.

![img_0031](https://user-images.githubusercontent.com/44229/50046604-b934e000-0073-11e9-9bb8-f7d972d6058b.jpg)
![img_0032](https://user-images.githubusercontent.com/44229/50046605-bf2ac100-0073-11e9-84fa-535de7301cc8.jpg)

**Comment by Dejal** [comment]
I can't reproduce this. I tried adding the World News Group site in the screenshot, but that didn't help either. Can you please reply to the customer and see if they have figured out any reproducible way to demo the issue?

**Comment by michaeldhopkins** [comment]
Hi, I don’t have a reproducible path, but it happens about once every time I go through my feed, so I’ll take notes and see if I can come up with something helpful.

**Comment by michaeldhopkins** [comment]
Okay this appears to be reproducible on my device:
1. Don’t use Newsblur for at least ~15 minutes (not sure how long exactly, and whether using other apps in the meantime triggers this, but I am usually in Safari.)
2. Open Newsblur while in landscape orientation. 
3. Visit this page from the feed in Newsblur: 
https://www.challies.com/a-la-carte/a-la-carte-january-30-5a/
4. Click the first link (Mary:...)
5. It should open with the black box.

It seems to be an issue with the first link opened after opening up Newsblur for the first time in awhile.

**Comment by Dejal** [comment]
Thanks for that, but it's still not occurring for me, even with that article and link. Do you use a mix of portrait and landscape, or only landscape?  I can't reproduce it in either orientation, or with orientation lock on or off.  It could be related to some difference in configuration.

The 15 minute thing is probably just NewsBlur being evicted from memory after you use other apps for a while; you could try force quitting it to have a similar effect.

It's possible that a screen recording showing it occurring may let me spot some clue, if you want to try that, but your steps seem clear enough.

**Comment by michaeldhopkins** [comment]
I only use landscape with the Smart Keyboard Folio attached. I’d be happy to try changing some settings or sending over some kind of diagnostic, if it’d help. 

I tried to narrow down the delay/quitting and found something interesting. If I force quit Newsblur and immediately open it again by tapping the dock icon, the issue won’t happen. If I force quit, then open any other app (tried Safari, Netflix and Slack), then either cmd+tab or tap the dock icon to switch to Newsblur, it will happen every time.

**Comment by Dejal** [comment]
This was tricky to reproduce, but thanks to your detailed description, I was able to do so.

My steps:

1. Use my iPad Pro in landscape orientation.
2. Force quit NewsBlur.
3. Open Safari and load a page.
4. Switch to NewsBlur via Cmd-Tab.
5. Go to the Tim Chaillies article you mentioned (in my Saved Stories).
6. Click the Mary link.
7. Wackiness ensues.

It doesn't occur when running from Xcode, but I was able to connect the debugger to the app once running, and saw that the `SFSafariViewController` was causing it when presented.  The UI orientation is changed to portrait, despite the device still being landscape, and not sending any orientation changed notifications.  I don't know why that page causes that misbehavior, but I suspect there's nothing I can do to stop it.

However, I am able to avoid the issue by using a page sheet presentation style, which displays the Safari view a little narrower when in landscape, but still plenty big enough. It's still full-screen when in portrait orientation, or on iPhones.
